### ISSUE TITLE
Llama 3.1 405B scan+offloading OOMs

### ISSUE THREAD

**Issue by tengyifei** [AI-Hypercomputer/torchprime#160]
Directly turning on scan+offloading in Llama 3.1 405B causes the training to OOM on the backward pass. The culprit is computing the gradient of the loss w.r.t. the weight of the `o_proj` layer. This gradient is only 1D sharded and not 2D sharded. That then infects a bunch of other ops to be also 1D sharded and causes the OOM.

https://github.com/AI-Hypercomputer/torchprime/commit/350f20e4abef199fa4b5ec68afea599dbe6ad5b8 contains an ugly hack to avoid the OOM.

We need to figure out how to avoid this to be able to achieve 32% MFU in Llama 3.1 405B via scan+offloading.

**Comment by tengyifei** [comment]
Fixed by https://github.com/pytorch/xla/pull/8879
